product-catalog/app/productcatalog/services/ProductProjectionServiceImpl.java

```diff
@@ -32,8 +36,16 @@ public ProductProjectionServiceImpl(final PlayJavaSphereClient sphere) {
     }
 
     public F.Promise<Optional<ProductProjection>> searchProductBySlug(final Locale locale, final String slug) {
-        return sphere.execute(ProductProjectionQuery.ofCurrent().bySlug(locale, slug))
+        final F.Promise<Optional<ProductProjection>> productOptionalPromise = sphere.execute(ProductProjectionQuery.ofCurrent().bySlug(locale, slug))
                 .map(PagedQueryResult::head);
+        productOptionalPromise.onRedeem(productOptional -> {
+            if (productOptional.isPresent()) {
+                LOGGER.trace("Found product for slug {} in locale {} with ID {}.", slug, locale, productOptional.get().getId());
+            } else {
+                LOGGER.trace("No product found for slug {} in locale {}.", slug, locale);
```

Comment:
`"... for slug " + slug + " ..."` instead of for slug `"... for slug {} ..."` and so on.